Source/engine/animationinfo.cpp

```diff
@@ -65,10 +65,13 @@ uint8_t AnimationInfo::getAnimationProgress() const
 	int32_t tickModifier = tickModifier_;
 
 	if (relevantFramesForDistributing_ <= 0) {
+		if (ticksPerFrame <= 0) {
+			Log("getAnimationProgress: Invalid ticksPerFrame {}", ticksPerFrame);
+		}
 		// This logic is used if animation distribution is not active (see getFrameToUseForRendering).
 		// In this case the variables calculated with animation distribution are not initialized and we have to calculate them on the fly with the given information.
-		ticksSinceSequenceStarted = ((currentFrame * ticksPerFrame) + tickCounterOfCurrentFrame) * baseValueFraction;
-		tickModifier = baseValueFraction / ticksPerFrame;
+		ticksSinceSequenceStarted = ((currentFrame * std::max(static_cast<int8_t>(1), ticksPerFrame)) + tickCounterOfCurrentFrame) * baseValueFraction;
+		tickModifier = baseValueFraction / std::max(static_cast<int8_t>(1), ticksPerFrame);

```

Comment:
```suggestion
		ticksSinceSequenceStarted = ((currentFrame * std::max<int8_t>(1, ticksPerFrame)) + tickCounterOfCurrentFrame) * baseValueFraction;
		tickModifier = baseValueFraction / std::max<int8_t>(1, ticksPerFrame);

```
you can force the use of a particular template instead of casting